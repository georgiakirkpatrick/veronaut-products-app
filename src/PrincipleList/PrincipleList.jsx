import React from 'react'
import Principle from '../Principle/Principle'
import './PrincipleList.css'

const PrincipleList = ({
  principles = [
    {
      id: 1,
        title: 'Good design', 
        description: 'We value versatile styles that bring out our best selves in the office and at happy hour.  That hold up on our bike commutes and school pick-ups.  Our ideal clothing is bra-friendly, easy care, and has pockets.',
        symbol: 'times'
    }
  ],
  readMore = ''
}) => {
  const generatePrinciples = principles.map(principle => (
    <Principle
      key={principle.id}
      title={principle.title} 
      description={principle.description}
      symbol={principle.symbol}
    />
  ))
  
  return (
    <section className='PrincipleList'>
      <header>
        <h2>
          Transparency is key
        </h2>

        <p>
          Veronaut helps you understand product descriptions and points out the information brands choose to omit.
        </p>
      </header>

      <ul>
        {generatePrinciples}
      </ul>

      {readMore}
    </section>
  )
}

export default PrincipleList;