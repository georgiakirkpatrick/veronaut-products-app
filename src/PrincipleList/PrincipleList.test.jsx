import React from 'react'
import { render, screen } from '@testing-library/react'
import PrincipleList from './PrincipleList'
import { describe, it, expect } from 'vitest'

const principles = [
  {
    id: 1,
    title: 'Good design', 
    description: 'We value versatile styles that bring out our best selves in the office and at happy hour.  That hold up on our bike commutes and school pick-ups.  Our ideal clothing is bra-friendly, easy care, and has pockets.',
    symbol: 'times'
  }
]

describe ('PrincipleList', () => {
  it ('renders without crashing', () => {
    render(<PrincipleList />)
    screen.debug()
  })

  it('renders the UI as expected', () => {
    const tree = render(<PrincipleList principles={principles} />)
    expect(tree).toMatchSnapshot()    
  })
})