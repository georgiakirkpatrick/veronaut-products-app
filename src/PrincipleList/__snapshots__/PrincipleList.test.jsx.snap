// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PrincipleList > renders the UI as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="PrincipleList"
      >
        <header>
          <h2>
            Transparency is key
          </h2>
          <p>
            Veronaut helps you understand product descriptions and points out the information brands choose to omit.
          </p>
        </header>
        <ul>
          <li
            class="Principle"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-times fa-lg "
              data-icon="times"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
            <h3>
              Good design
            </h3>
            <p>
              We value versatile styles that bring out our best selves in the office and at happy hour.  That hold up on our bike commutes and school pick-ups.  Our ideal clothing is bra-friendly, easy care, and has pockets.
            </p>
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="PrincipleList"
    >
      <header>
        <h2>
          Transparency is key
        </h2>
        <p>
          Veronaut helps you understand product descriptions and points out the information brands choose to omit.
        </p>
      </header>
      <ul>
        <li
          class="Principle"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-times fa-lg "
            data-icon="times"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
              fill="currentColor"
            />
          </svg>
          <h3>
            Good design
          </h3>
          <p>
            We value versatile styles that bring out our best selves in the office and at happy hour.  That hold up on our bike commutes and school pick-ups.  Our ideal clothing is bra-friendly, easy care, and has pockets.
          </p>
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
